private/model/api/endpoint_arn.go

```diff
@@ -74,13 +76,18 @@ const accountIDWithARNShapeTmplDef = `
 {{ range $_, $name := $.MemberNames -}}
 	{{ $elem := index $.MemberRefs $name -}}
 	{{ if $elem.AccountIDMemberWithARN -}}
-		func (s *{{ $.ShapeName }}) updateAccountID(accountId string) error {
+		// updateAccountID returns a pointer to a modified copy of input, 
+		// if account id is not provided, we update the account id in modified input
+		// if account id is provided, but doesn't match with the one in ARN, we throw an error
+		// if account id is not updated, we return nil. Note that original input is not modified. 
+		func (s {{ $.ShapeName }}) updateAccountID(accountId string) (interface{}, error) {
 			if s.{{ $name }} == nil {
-				s.{{ $name }} = aws.String(accountId)	
+				s.{{ $name }} = aws.String(accountId)
+				return &s, nil
 			} else if *s.{{ $name }} != accountId  {
-				return fmt.Errorf("Account ID mismatch, the Account ID cannot be specified in an ARN and in the accountId field")
+				return &s, fmt.Errorf("Account ID mismatch, the Account ID cannot be specified in an ARN and in the accountId field")
 			}
-			return nil 
+			return nil, nil
 		}
```

Comment:
Should this copy the shape before assigning the update to it, then return the copy?

Reply:
Missed this was changed to value instead of pointer receiver.

Reply:
Discussed offline. The function receiver is changed from pointer to value receiver. So no need to explicitly copy.